### PR TITLE
Check AWS IAM Key age

### PR DESCRIPTION
**Pull request by erino**
This alerts if AWS IAM Keys are older then 90 days. AWS Trusted Advisor recommends that IAM keys are rotated every 90 days, this alert will show keys that are older than this.

### CODE REVIEW

File: modules/monitoring/templates/check_aws_iam_key_age.cfg.erb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+define command {
+    command_name check_aws_iam_key_age
+    command_line /usr/lib/nagios/plugins/check_aws_aws_iam_key_age --key <%= @access_key %> --secret <%= @secret_key %> --region <%= @region %> --days <%= @max_age %>
+}
```
Comment (cbaines) [inline modules/monitoring/templates/check_aws_iam_key_age.cfg.erb]: The trailing newline is missing for this file.

File: modules/monitoring/manifests/checks/aws_iam_key.pp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#   Should we enable the monitoring check?
+#
+class monitoring::checks::aws_iam_key (
+  $region = undef,
+  $access_key = undef,
+  $secret_key = undef,
+  $enabled = true,
+) {
+  if $enabled and $::aws_migration {
+    $max_age = 365
```
Comment (cbaines) [inline modules/monitoring/manifests/checks/aws_iam_key.pp]: There's a greater possibility for confusion with the value here, and a default in the script. Could the value just be configured through changing the script?

Comment (erino) [reply modules/monitoring/manifests/checks/aws_iam_key.pp]: I wasn't sure the best way of doing this. I want to gradually decrease the max age as there are currently lots of keys that need rotation.

File: modules/monitoring/templates/check_aws_iam_key_age.cfg.erb (hunk 1 above)
Comment (erino) [reply modules/monitoring/templates/check_aws_iam_key_age.cfg.erb]: Fixed in 8961079 via a05475b

File: hieradata/integration.yaml [hunk 3]
```diff
@@ -170,6 +170,7 @@ monitoring::checks::aws_origin_domain: "integration.govuk.digital"
 monitoring::checks::sidekiq::enable_signon_check: false
 monitoring::checks::pingdom::enable: false
 monitoring::checks::ses::region: eu-west-1
+monitoring::checks::aws_iam_key: 'eu-west-1'
```
[the remark this answers is not in the document]
Comment (erino) [reply hieradata/integration.yaml]: Fixed in 99a9e86

File: modules/monitoring/manifests/checks/aws_iam_key.pp (hunk 2 above)
Comment (cbaines) [reply modules/monitoring/manifests/checks/aws_iam_key.pp]: It'll work as is, this was just a minor though I had. So this isn't something that I think needs changing now.

**Pull request merged**